### PR TITLE
Document System.Reflection

### PR DESCRIPTION
**Pull request by carlossanlop**
Area owners: @steveharter @GrabYourPitchforks . Can you please take a quick look? I based my documentation on the source code.

### CODE REVIEW

File: xml/System.Reflection/ReflectionTypeLoadException.xml [hunk 1]
```diff
@@ -307,8 +307,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns the fully qualified name of this exception and the messages for all the loader exceptions.</summary>
+        <returns>A string containing the fully qualified name of this exception and the exception messages for the its loader exceptions.</returns>
```
Comment (rpetrusha) [inline xml/System.Reflection/ReflectionTypeLoadException.xml]:
```suggestion
        <returns>A string containing the fully qualified name of this exception and the exception messages for its loader exceptions.</returns>
```

Comment (carlossanlop) [reply xml/System.Reflection/ReflectionTypeLoadException.xml]: Ah thanks for catching it. I thought this was going to be a flawless PR :(

**Comment by rpetrusha** [comment]
Thanks, @carlossanlop. I'll merge when the build completes successfully.

File: xml/System.Reflection/MethodImplAttributes.xml [hunk 2]
```diff
@@ -162,7 +162,7 @@
       </ReturnValue>
       <MemberValue>512</MemberValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Specifies that the method may contain hot code and should be aggressively optimized.</summary>
```
Comment (steveharter) [inline xml/System.Reflection/MethodImplAttributes.xml]: Value 256 states
`Specifies that the method should be inlined wherever possible`.

For consistency should we just state fot 512:
`Specifies that the method should be optimized wherever possible`

File: xml/System.Reflection/ReflectionTypeLoadException.xml [hunk 3]
```diff
@@ -307,8 +307,8 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Returns the fully qualified name of this exception and the messages for all the loader exceptions.</summary>
+        <returns>A string containing the fully qualified name of this exception and the exception messages for its loader exceptions.</returns>
         <remarks>To be added.</remarks>
```
Comment (steveharter) [inline xml/System.Reflection/ReflectionTypeLoadException.xml]: Do you want to remove this line ("To be added.")

File: xml/System.Reflection/TypeDelegator.xml [hunk 4]
```diff
@@ -1619,8 +1619,8 @@ t.InvokeMember ("MyArray", BindingFlags.SetField, null, this, new
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets the <see cref="P:System.Reflection.MemberInfo.IsCollectible" /> value for this object's <see cref="F:System.Reflection.TypeDelegator.typeImpl" />, which indicates whether this object, which is a <see cref="T:System.Reflection.MemberInfo" /> implementation, is part of an assembly held in a collectible <see cref="T:System.Runtime.Loader.AssemblyLoadContext" />.</summary>
+        <value><see langword="true" /> if this object, which is a <see cref="T:System.Reflection.MemberInfo" /> implementation, is part of an assembly held in a collectible assembly load context; otherwise, <see langword="false" />.</value>
         <remarks>To be added.</remarks>
```
Comment (steveharter) [inline xml/System.Reflection/TypeDelegator.xml]: Remove?

File: xml/System.Reflection/TypeInfo.xml [hunk 5]
```diff
@@ -94,7 +94,7 @@
       </AssemblyInfo>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Initializes a <see cref="T:System.Reflection.TypeInfo" /> instance.</summary>
         <remarks>To be added.</remarks>
```
Comment (steveharter) [inline xml/System.Reflection/TypeInfo.xml]: Remove?

File: xml/System.Reflection/ReflectionTypeLoadException.xml (hunk 3 above)
Comment (carlossanlop) [reply xml/System.Reflection/ReflectionTypeLoadException.xml]: No, that string is expected by the Docs repo for documentation that is empty.

File: xml/System.Reflection/MethodImplAttributes.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Reflection/MethodImplAttributes.xml]: Thanks. I will submit them in a new PR. This has already been commited.

File: xml/System.Reflection/TypeDelegator.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.Reflection/TypeDelegator.xml]: No. It's expected for empty remarks.

File: xml/System.Reflection/TypeInfo.xml (hunk 5 above)
Comment (carlossanlop) [reply xml/System.Reflection/TypeInfo.xml]: No, expected for empty remarks.

**Pull request merged**